src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -1327,6 +1331,99 @@ class spell_pri_power_word_shield : public SpellScriptLoader
+
+        class spell_pri_angelic_feather_trigger_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_pri_angelic_feather_trigger_SpellScript);
+
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                if (!sSpellMgr->GetSpellInfo(SPELL_PRIEST_ANGELIC_FEATHER_AREATRIGGER))
+                    return false;
+                return true;
+            }
+
+            void HandleEffectDummy(SpellEffIndex /*effIndex*/)
+            {
+                Position destPos = GetHitDest()->GetPosition();
+
+                SpellCastTargets targets;
+                targets.SetDst(destPos);
+                GetCaster()->CastSpell(targets, sSpellMgr->GetSpellInfo(SPELL_PRIEST_ANGELIC_FEATHER_AREATRIGGER), nullptr);
+            }
+
+            void Register() override
+            {
+                OnEffectHit += SpellEffectFn(spell_pri_angelic_feather_trigger_SpellScript::HandleEffectDummy, EFFECT_0, SPELL_EFFECT_DUMMY);
+            }
+        };
+
+        SpellScript* GetSpellScript() const override
+        {
+            return new spell_pri_angelic_feather_trigger_SpellScript();
+        }
+};
+
+// Angelic Feather areatrigger - created by SPELL_PRIEST_ANGELIC_FEATHER_AREATRIGGER
+class areatrigger_pri_angelic_feather : public AreaTriggerEntityScript
+{
+public:
+    areatrigger_pri_angelic_feather() : AreaTriggerEntityScript("areatrigger_pri_angelic_feather") { }
+
+    struct areatrigger_pri_angelic_featherAI : AreaTriggerAI
+    {
+        areatrigger_pri_angelic_featherAI(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger) { }
+
+        // Called when the AreaTrigger has just been initialized, just before added to map
+        void OnInitialize() override
+        {
+            if (Unit* caster = at->GetCaster())
+            {
+                std::vector<AreaTrigger*> areaTriggers = caster->GetAreaTriggers(SPELL_PRIEST_ANGELIC_FEATHER_AREATRIGGER);
+
+                if (areaTriggers.size() >= 2)
+                    areaTriggers.front()->SetDuration(0);
+
+                // Caster is prioritary
+                if (caster->IsWithinDist(at, at->GetTemplate()->SphereDatas.Radius))
+                    OnUnitEnter(caster);
+            }
+        }
+
+        void OnUnitEnter(Unit* unit) override
+        {
+            if (Unit* caster = at->GetCaster())
+            {
+                if (caster->IsFriendlyTo(unit))
+                {
+                    // If target already has aura, increase duration to max 130% of initial duration
+                    if (Aura* aura = unit->GetAura(SPELL_PRIEST_ANGELIC_FEATHER_AURA))
+                    {
+                        int32 newDuration = aura->GetDuration() + aura->GetMaxDuration();
+                        int32 maxPossibleNewDuration = CalculatePct(aura->GetMaxDuration(), 130);
+
+                        aura->SetDuration(newDuration > maxPossibleNewDuration ? maxPossibleNewDuration : newDuration);
```

Comment:
nvm, my mistake, the original blue post sounds like this:
"Recasting periodic damage over time and healing over time effects that are already on the target now extends those effects to up to 130% of the normal duration of the effect."
keep the code, my previous comment does not apply here